shard/src/ext/purge_command.py

```diff
@@ -1,41 +1,192 @@
+import discord
 from discord.ext import commands
-from contextlib import suppress
 
+import re
+from datetime import datetime, timedelta
 
-@commands.command()
-async def purge(ctx, *args):
+TIME_REGEX = re.compile(r"(\d+)([m|s])")
+
+
+@commands.group()
+async def purge(ctx):
```

Comment:
I'm not sure exactly how it works with subcommands, but only admin should be able to run this

Reply:
The bot does currently check if an admin is running the command in the subcommands so there aren't any actual permission issues. but I can move that into the actual purge command if you want me to